### DISCUSSION TITLE
Unable to get ResponseWriter from Decode or endpoint.

### DISCUSSION THREAD

**Discussion by hecomp** [go-kit/kit#1156] [Q&A]
Hi, by convertion of the package, Decode only has context and http.Request. So, how can we get ResponseWriter from endpoint or Decode so we can decode the body and do some validation. 

Were looking for something like:

`func Decode(cxt context.Context, w http.ResponseWriter, r *http.Request) (interface{}, error) {}`

**Comment by peterbourgon** [comment]
Why do you want Decode to have access to the ResponseWriter? It's only responsibility is to decode the request.

**Comment by phper-chen** [comment]
u can add a middleware after the biz-logic fn
